internal/controller/state_mapper.go

```diff
@@ -0,0 +1,144 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the MIT License.
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2024 Datadog, Inc.
+
+package controller
+
+import (
+	"github.com/DataDog/temporal-worker-controller/api/v1alpha1"
+	"github.com/DataDog/temporal-worker-controller/internal/k8s"
+	"github.com/DataDog/temporal-worker-controller/internal/temporal"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// StateMapper maps between Kubernetes and Temporal states
+type StateMapper struct {
+	k8sState      *k8s.DeploymentState
+	temporalState *temporal.TemporalWorkerState
+}
+
+// NewStateMapper creates a new state mapper
+func NewStateMapper(k8sState *k8s.DeploymentState, temporalState *temporal.TemporalWorkerState) *StateMapper {
```

Comment:
Since these symbols don't seem to be used outside the package, I think they can be made private.

Not a big deal since it's already under /internal, but still good hygiene in case anything here is ever refactored into a public package.